Card images: animate on load and on hover

cameralibre opened the pull request:
You can see both of these effects on https://the-lumery-staging.webflow.io/projects

Originally I had painstakingly created both of these animations using Webflow interactions, but it was very buggy and eventually stopped working for some reason, I couldn't work out why and didn't have any way of debugging it. 
So... I had to recreate the animations using something more reliable. (Web Animations API)

liamjohnston commented:
I'm not seeing the animations at that URL? 🤔

liamjohnston commented on client-projects/the-lumery/cardImages.ts:
```diff
@@ -0,0 +1,107 @@
+
+  cards.forEach((card) => {
+    const imageContainer = card.querySelector(IMAGE_CONTAINER_SELECTOR);
+    if (!(imageContainer instanceof HTMLDivElement)) {
```
That's a nice check.

cameralibre replied:
yeah, I had been finding myself adding an `: HTMLDivElement | null` annotation to `const imageContainer` in order to use `imageContainer.style`, and then _also_ checking `if (imageContainer)`... and I remembered there's a way I could achieve both in one check!

liamjohnston replied:
Yeah, makes sense. I'll have to remember this one.

liamjohnston commented on client-projects/the-lumery/index.ts:
```diff
@@ -44,6 +46,13 @@ document.addEventListener("DOMContentLoaded", () => {
 
+  // images within cards fade up and on when they appear on screen, and have a
+  // hover effect applied.
+  if (document.querySelectorAll(CARD_LINK_SELECTOR)) {
```
I think this will always be truthy, even if no items are found. You might need to specifically check for the `length`?

cameralibre replied:
ah yep you're right

liamjohnston reviewed (approved):
that's good codin' 👨‍🍳👌

cameralibre commented:
you should be able to see the animations at that url now! we had switched it to getting its JS from `localhost` while we incorporated Cliff's work. Just switched it back.

liamjohnston commented:
Ah cool. Yep, works perfectly 👍
